gapil/runtime/cc/replay/dataex.h

```diff
@@ -19,6 +19,17 @@
 #include "core/cc/interval_list.h"
 
 #include <unordered_map>
+#include <vector>
+
+namespace gapil {
+namespace runtime {
+namespace replay {
+
+// align returns val aligned up the the next multiple of by.
```

Comment:
Just as a note, if these are always power-of-2, this can be simplified

return (val + (by-1))&(by-1);

If this is being run a lot idiv is not friendly on Arm.

Reply:
Ack will optimize if I see it show up in a profile.